mw_organization_activity_test.go

```diff
@@ -221,17 +207,9 @@ func TestProcessRequestLiveRedisRollingLimiter(t *testing.T) {
 
 func TestProcessRequestOffThreadRedisRollingLimiter(t *testing.T) {
 	// setup global config
-	enforceOrgQuotas := config.Global.EnforceOrgQuotas
 	config.Global.EnforceOrgQuotas = true
-	enableRedisRollingLimiter := config.Global.EnableRedisRollingLimiter
 	config.Global.EnableRedisRollingLimiter = true
-	experimentalProcessOrgOffThread := config.Global.ExperimentalProcessOrgOffThread
 	config.Global.ExperimentalProcessOrgOffThread = true
-	defer func() {
-		config.Global.EnforceOrgQuotas = enforceOrgQuotas
-		config.Global.EnableRedisRollingLimiter = enableRedisRollingLimiter
-		config.Global.ExperimentalProcessOrgOffThread = experimentalProcessOrgOffThread
-	}()
```

Comment:
I think you need to call `defer resetTestConfig()` after each config declaration section, in both tests, similar to how its done in the rest of tests.

Reply:
I've tried different approaches including this one but it introduces data races.
I think the problem is that by the time we change `global.Config` in deferred function - there are still some go-routines running and reading config (like ones launched in `ProcessRequestOffThread()`)

Reply:
For cases with config races, I introduced new mutex `configMu` you can see its usage.

Reply:
yeah, I saw it, just thinking about how feasible would be to introduce new locks in middleware code just because config is being changed in tests. how often do we expect config to be change during gateway run?

I feel like those locks are extra as we change config structure by changing file and restarting actual GW-process